snapcraft/_baseplugin.py

```diff
@@ -199,10 +200,20 @@ def run(self, cmd, cwd=None, **kwargs):
             cwd = self.builddir
         print(' '.join(cmd))
         os.makedirs(cwd, exist_ok=True)
-        return common.run(cmd, cwd=cwd, **kwargs)
+        try:
+            return common.run(cmd, cwd=cwd, **kwargs)
+        except CalledProcessError as process_error:
+            raise errors.SnapcraftPluginCommandError(
+                command=cmd, part_name=self.name,
+                exit_code=process_error.returncode)
```

Comment:
Agh sorry, missed this before, but you probably want a `from process_error` on both of these.